src/pal/src/map/map.cpp

```diff
@@ -2241,7 +2241,9 @@ MAPmmapAndRecord(
     PAL_ERROR palError = NO_ERROR;
     LPVOID pvBaseAddress = NULL;
 
-    pvBaseAddress = mmap(addr, len, prot, flags, fd, offset);
+    off_t adjust = offset & (VIRTUAL_PAGE_MASK);
+
+    pvBaseAddress = mmap(static_cast<char *>(addr) - adjust, len + adjust, prot, flags, fd, offset - adjust);
```

Comment:
@sdmaclea I am sorry, I've missed the MAP_PRIVATE flag presence, so you are right and there is no security hole.
As for the amd64 issue I also missed the MAP_FIXED flag, so you are right too that it is not causing the issue I was referring to. The mappings of multiple consecutive sections that are not page aligned would just overlap. But that brings another concern for all targets. What if these sections have different protection? Then it seems that after the file is mapped, all sections that map into a single virtual address space page would have the protection of the last section on the page. And what if one of these sections except the last one was executable? Wouldn't that kill its executability? And vice versa. This issue would not reproduce in 100% cases since it depends on the actual sizes and order of the sections. Have you tried to run CoreFX tests with your change on amd64? I guess that would have a higher chance of reproducing the issue.